frontend/src/metabase/collections/components/CollectionCopyEntityModal.jsx

```diff
@@ -30,6 +30,21 @@ function CollectionCopyEntityModal({
   onSaved,
   triggerToast,
 }) {
+  const [isShallowCopy, setIsShallowCopy] = useState(true);
+  const handleValuesChange = ({ is_shallow_copy }) => {
+    setIsShallowCopy(is_shallow_copy);
+  };
+
+  const getTitle = () => {
```

Comment:
nit: I'd love to move this function outside of the component to make its dependencies more explicit